File: Java/Src/Car/Car.java
package car;

public class Car {
    private CarState state;

    public void setState(CarState state) {
        this.state = state;
    }

    public void performStateAction() {
        if (state != null) {
            state.handleState(this);
        } else {
            System.out.println("Car state is undefined.");
        }
    }
}

File: Python/src/car/car.py
class Car:
    def __init__(self):
        self.state = None

    def set_state(self, state):
        self.state = state

    def perform_state_action(self):
        if self.state:
            self.state.handle_state(self)
        else:
            print("Car state is undefined.")
